drf/university/views.py

```diff
@@ -1,14 +1,35 @@
 from .serializers import UniversitySerializer
+from school.serializers import SchoolSerializer
 from .models import University
 from rest_framework.generics import (ListCreateAPIView,
-                                     RetrieveUpdateDestroyAPIView)
+                                     RetrieveUpdateDestroyAPIView,
+                                     ListAPIView)
 
 
 class UniversityListCreate(ListCreateAPIView):
     queryset = University.objects.all()
     serializer_class = UniversitySerializer
 
 
-class UniversityRetrieveUpdateDestroy(RetrieveUpdateDestroyAPIView):
+class UniversityDetail(RetrieveUpdateDestroyAPIView):
     queryset = University.objects.all()
     serializer_class = UniversitySerializer
+
+
+class UniversitySchoolsList(ListCreateAPIView):
+    serializer_class = SchoolSerializer
+
+    def get_queryset(self):
+        university = University.objects.get(
+            pk=self.kwargs.get('pk'))
+        return university.schools.all()
+
+
+class UniversitySchoolDetails(RetrieveUpdateDestroyAPIView):
+    serializer_class = SchoolSerializer
+
+    def get_queryset(self):
+        university = University.objects.get(
```

Comment:
Same thing. _get_ evaluates query (hits the database). Get queryset should return queryset not object from DB.